InnerEye/ML/configs/classification/CovidModel.py

```diff
@@ -234,24 +246,38 @@ def get_output_csv_path(mode: ModelExecutionMode) -> Path:
             p = get_best_epoch_results_path(mode=mode, model_proc=model_proc)
             return self.outputs_folder / p / MODEL_OUTPUT_CSV
 
+        def get_labels_and_predictions(df: pd.DataFrame) -> pd.DataFrame:
+            labels = []
+            predictions = []
+            for i, target in enumerate(self.target_names):
+                predictions.append(df[df[LoggingColumns.Hue.value] == target][LoggingColumns.ModelOutput.value].item())
+                labels.append(df[df[LoggingColumns.Hue.value] == target][LoggingColumns.Label.value])
+
+            return pd.DataFrame.from_dict({LoggingColumns.Patient.value: [df.iloc[0][LoggingColumns.Patient.value]],
+                                           LoggingColumns.ModelOutput.value: [np.argmax(predictions)],
+                                           LoggingColumns.Label.value: [np.argmax(labels)]})
+
+        def get_accuracy(df: pd.DataFrame) -> float:
+            df = df.groupby(LoggingColumns.Patient.value, as_index=False).apply(get_labels_and_predictions).reset_index(
+                drop=True)
+            df["tp+tn"] = df.apply(
+                lambda x: 1 if x[LoggingColumns.ModelOutput.value] == x[LoggingColumns.Label.value] else 0, axis=1)
+            return np.sum(df["tp+tn"].values) / len(df)
```

Comment:
Boolean `pd.Series` support arithmetic operations, so you can just do e.g. `(df[A] == df[B]).mean()`.